cpp/test/CMakeLists.txt

```diff
@@ -16,7 +16,7 @@
 
 function(ConfigureTest)
 
-  set(options OPTIONAL NCCL CUMLPRIMS MPI ML_INCLUDE)
```

Comment:
The more I think about this, I believe we are still going to need to link against NCCL to use `raft/comms/mpi_comms.hpp` but only for the mg tests and nowhere else. The `mpi_comms.hpp` are essentially creating a NCCL cluster using MPI and then handing off to NCCL everywhere (think of it like replacing Dask w/ MPI). 

Maybe we could use the `raft::distributed` COMPONENT for that? (or maybe not, since it also brings in UCX...).

Reply:
Components are neat, added `raft::distributed` linkage and notes of added dependencies (MPI, NCCL) for C++ MG tests

Reply:
Cuml's MG googletests should be using MPI, however they should still be tied to the comms interface so shouldn't need to know NCCL exists at all (this NCCL can be provided transitively through the raft::distributed component). Note also that wherher the underlying MPI configuration has been configured with UCX underneath is up to the user running the tests (so the raft MPI communicator will use ucx by proxy  and not directly) 

Cuml's mg pytests all use raft-dask.

Reply:
Right the python tests are fine. I guess the difference is that with the pytests raft-dask is a Python package whose underlying extension modules are linked to NCCL, and then cuml Python only interacts with it at the Python layer, whereas the cuml C++ tests are either linking directly to libraft or building against raft's headers, and in either case the distributed components are not precompiled anyway so there's always NCCL linking required?

Reply:
Since we are using PIMPL for the comms, the mg tests shouldnt need nccl during compile time. But yes, the mg tests binary will need to have raft::distributed on the link libraries (which should pass nccl through)

Reply:
Yup OK cool. makes sense. Dante has already made the necessary change here by adding the distributed component in get_raft.cmake, so I think we're good here.